main/php_variables.c

```diff
@@ -541,37 +541,56 @@ static zend_always_inline int valid_environment_name(const char *name, const cha
 	return 1;
 }
 
-void _php_import_environment_variables(zval *array_ptr)
+static zend_always_inline void import_environment_variable(HashTable *ht, char *env)
 {
-	char **env, *p;
+	char *p;
 	size_t name_len, len;
 	zval val;
 	zend_ulong idx;
 
+	p = strchr(env, '=');
+	if (!p
+		|| p == env
+		|| !valid_environment_name(env, p)) {
+		/* malformed entry? */
+		return;
+	}
+	name_len = p - env;
+	p++;
+	len = strlen(p);
+	if (len == 0) {
+		ZVAL_EMPTY_STRING(&val);
+	} else if (len == 1) {
+		ZVAL_INTERNED_STR(&val, ZSTR_CHAR((zend_uchar)*p));
+	} else {
+		ZVAL_NEW_STR(&val, zend_string_init(p, len, 0));
+	}
+	if (ZEND_HANDLE_NUMERIC_STR(env, name_len, idx)) {
+		zend_hash_index_update(ht, idx, &val);
+	} else {
+		php_register_variable_quick(env, name_len, &val, ht);
+	}
+}
+
+void _php_import_environment_variables(zval *array_ptr)
+{
+#ifndef PHP_WIN32
+	char **env;
+#else
+	char *environment, *env;
+#endif
```

Comment:
You can move these declarations into the respective #if blocks below.

Reply:
Oh, indeed, that would work fine for PHP-7.3. However, in PHP-7.4 there is
https://github.com/php/php-src/blob/038ca4bb07abc87f5030c0257fe81876799e22b4/main/php_variables.c#L550
between the declarations and the `for` loop. :(

Reply:
Oh well, we can do it just for master then, where we officially use C99.

Reply:
Makes sense. Done.